components/automate-gateway/gateway/middleware/auth_interceptor_internal_test.go

```diff
@@ -74,6 +111,82 @@ func TestHeaderAuthValidForClientAndPeer(t *testing.T) {
 	}
 }
 
+func TestComboAuth(t *testing.T) {
+	albCert, _ := devCertToEncodedAndPeer(t, "automate-load-balancer")
+	agCert, agPeer := devCertToEncodedAndPeer(t, "automate-gateway")
+	otherServiceCert, _ := devCertToEncodedAndPeer(t, "deployment-service")
+
+	reqFromALBCtx := metadata.NewIncomingContext(
+		peer.NewContext(context.Background(), agPeer),
+		metadata.Pairs("x-client-cert", albCert, "grpcgateway-x-client-cert", agCert,
+			runtime.MetadataPrefix+"projects", "project1,project2",
+		))
+
+	reqDirectToGatewayCtx := metadata.NewIncomingContext(
+		peer.NewContext(context.Background(), agPeer),
+		metadata.Pairs("x-client-cert", albCert, "grpcgateway-x-client-cert", otherServiceCert,
+			runtime.MetadataPrefix+"projects", "project1,project2",
+		))
+
+	t.Run("It returns error for authn failure", func(t *testing.T) {
+		a := authInterceptor{
+			authn: testAuthnClient{err: errors.New("sample authn failure")},
+			authz: testAuthzHandler{},
+		}
+		_, err := a.combinedAuth(reqFromALBCtx, nil)
+		require.Error(t, err)
```

Comment:
[nit] this should be an `assert` since it's testing for the error

Reply:
The actual `err` here is the `errors.New("sample authn failure")` from above because the combo auth function is just passing the error through. So we have to trust that the authenticator is doing the right thing.

Reply:
That said, I added a check to assert that we don't break something that would otherwise be working (e.g., by using `errors.Wrap`) in the `combinedAuth` method, below.